### PR TITLE
Microoptimizations

### PR DESCRIPTION
**Pull request by roman-khimov**
These don't change a lot, but still they're good and and they're local, so shouldn't be a problem.

### CODE REVIEW

File: pkg/core/native/native_gas.go [hunk 1]
```diff
@@ -54,13 +54,20 @@ func newGAS(init int64) *GAS {
...
 	}
 	if sign := amount.Sign(); sign == 0 {
-		return nil
+		// Requested self-transfer amount can be higher than actual balance.
+		if checkBal != nil {
+			if acc.Balance.Cmp(checkBal) < 0 {
+				err = errors.New("insufficient funds")
+			}
+		}
+		*si = acc.Bytes()
```
Comment (AnnaShaleva) [inline pkg/core/native/native_gas.go:69]: Why do we need this line?
```
*si = acc.Bytes()
```

Comment (roman-khimov) [reply pkg/core/native/native_gas.go:69]: Yeah, I've noticed that NEO does it in `increaseBalance`, but I've also questioned myself on why it does it. Zero balance? But it shouldn't be stored anyway. It needs to be checked, maybe it's not needed.

Comment (roman-khimov) [reply pkg/core/native/native_gas.go:69]: It was added with 7987cdadc, so it's important for NEO, but it actually can be omitted for GAS (as it only stores balance).

Comment (roman-khimov) [reply pkg/core/native/native_gas.go]: Fixed.

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**